src/LocationManager.swift

```diff
@@ -10,22 +10,36 @@ import UIKit
 import CoreLocation
 import OnTheRoad
 
-extension OTRGeoPoint {
-  
-}
-
+/// LocationManagerDelegate Protocol
 @objc public protocol LocationManagerDelegate {
+
+  /// The user authorized the application to receive location updates
   @objc optional func authorizationDidSucceed()
+
+  /// The user denied the application from receiving location updates.
   @objc optional func authorizationDenied()
+
+  /// The user is restricted from authorizing the application for location updates. This is normally due to parental control lockout.
   @objc optional func authorizationRestricted()
+
+  /// The device received a new location.
   @objc optional func locationDidUpdate(_ location: CLLocation)
 }
 
-
+/**
+ The `LocationManager` class is a wrapper around iOS's built in location subsystem. It provides a simpler interface and customization options.
+ 
+ - Note: The class is expected to be used as a singleton. As such, access to the class is via the `sharedManager` object.
```

Comment:
Is there anyway to enforce the singleton pattern in Swift for example with a private constructor?